gammapy/scripts/cta_irf.py

```diff
@@ -95,3 +102,219 @@ def read(cls, filename):
+        energy_hi = table['{}_HI'.format(energy_col)].quantity
+        energy = np.append(
+            energy_lo.value, energy_hi[-1].value) * energy_lo.unit
+        data = table['{}'.format(data_col)].quantity
+        return cls(energy=energy, data=data)
+
+    def plot(self, ax=None, energy=None, **kwargs):
+        """Plot point spread function
+
+        Parameters
+        ----------
+        ax : `~matplotlib.axes.Axes`, optional
+            Axis
+        energy : `~astropy.units.Quantity`
+            Energy nodes
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`
+            Axis
+
+        """
+        import matplotlib.pyplot as plt
+        ax = plt.gca() if ax is None else ax
+
+        if energy is None:
+            energy = self.energy.nodes
+        values = self.evaluate(energy=energy)
+        xerr = (energy.value - self.energy.data[:-1].value,
+                self.energy.data[1:].value - energy.value)
+        ax.errorbar(energy.value, values.value, xerr=xerr, fmt='o', **kwargs)
+        ax.set_xscale('log')
+        ax.set_xlabel('Energy [{}]'.format(self.energy.unit))
+        ax.set_ylabel(
+            'Angular resolution 68 % containment [{}]'.format(self.data.unit)
+        )
+
+        return ax
+
+
+class CTAPerf(object):
+    """CTA instrument response function container.
+
+    Class handling CTA performance.
+
+    For now we use the production 2 of the CTA IRF
+    (https://portal.cta-observatory.org/Pages/CTA-Performance.aspx)
+
+    The IRF format should be compliant with the one discussed
+    at http://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/.
+    Work will be done to handle better the PSF and the background rate.
+
+    This class is similar to `~gammapy.data.DataStoreObservation`,
+    but only contains performance (no event data or livetime info).
+    TODO: maybe re-factor code somehow to avoid code duplication.
+
+    Parameters
+    ----------
+    aeff : `~gammapy.irf.EffectiveAreaTable`
+        Effective area
+    edisp : `~gammapy.irf.EnergyDispersion`
+        Energy dispersion
+    psf : `~gammapy.scripts.Psf68Table`
+        Point spread function
+    bkg : `~gammapy.scripts.BgRateTable`
+        Background rate
+    """
+
+    def __init__(self, aeff=None, edisp=None, psf=None, bkg=None):
+        self.aeff = aeff
+        self.edisp = edisp
+        self.psf = psf
+        self.bkg = bkg
+
+    @classmethod
+    def read(cls, filename):
+        """
+        Read from a FITS file.
+
+        Parameters
+        ----------
+        filename : `str`
+            File containing the IRFs
+        """
+        filename = str(make_path(filename))
+
+        aeff = EffectiveAreaTable.read(filename, hdu='SPECRESP')
+
+        # Get EnergyDispersion from HDUList
+        hdu_list = fits.open(filename)
+        edisp = EnergyDispersion.from_hdulist(hdu_list=hdu_list)
+
+        # Do not work, can't understand why, see BgRateTable and Psf68Table class
```

Comment:
The problem is the following:

BgRateTable.read() is implemented [here](http://docs.gammapy.org/en/latest/_modules/gammapy/utils/nddata.html#NDDataArray.read). As you see it calls
```
fits.open(*args, **kwargs)
```
However, ``hdu`` is not a valid kwarg. ``fits.open`` always returns the entire HDUList. For the effective are it works by change because it is stored in the first HDU. This behaviour of ``NDDataArray.read`` is obviously very bad.

But for now I propose to use the solution you found here, and change it as soons as the implementation of ``NDDataArray`` has improved. There is already an issue about the API of ``read`` but I can't find it at the moment.